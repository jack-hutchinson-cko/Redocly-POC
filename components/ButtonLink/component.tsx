import React from 'react';
import styled from 'styled-components';

const StyledLink = styled.a`
	background: ${(props: { hierarchy: Hierarchy }) =>
		props.hierarchy === 'primary' ? 'var(--btn-primary-bg)' : 'var(--btn-secondary-bg)'};
	border: none;
	border-radius: 8px;
	color: ${(props: { hierarchy: Hierarchy }) =>
		props.hierarchy === 'primary' ? 'var(--btn-primary-font)' : 'var(--btn-secondary-font)!important'};
	display: inline-block;
	font-size: 16px;
	font-weight: 500;
	line-height: 24px;
	padding: 12px 24px;
	text-align: center;
	text-decoration: none;
	transition: background 0.5s, color 0.5s;
	width: max-content;

	:hover {
		background: ${(props: { hierarchy: Hierarchy }) =>
			props.hierarchy === 'primary' ? 'var(--btn-primary-hover)' : 'var(--btn-secondary-hover)'};
	}
`;

type Hierarchy = 'primary' | 'secondary';

interface Props {
	children: string;
	hierarchy?: Hierarchy;
	to: string;
}

export const ButtonLink = ({ children, hierarchy = 'primary', to }: Props): JSX.Element => (
	<StyledLink href={to} hierarchy={hierarchy}>
		{children}
	</StyledLink>
);

export default ButtonLink;
