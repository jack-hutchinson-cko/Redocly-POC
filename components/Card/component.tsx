import React from 'react';
import styled from 'styled-components';
import { CardArrow } from '../svg';

const StyledContainer = styled.a`
	background: transparent;
	border-radius: 8px;
	border: 1px solid var(--border);
	box-shadow: rgb(0 18 44 / 15%) 0px 1px 3px;
	color: var(--font-color);
	display: flex;
	flex-direction: column;
	padding: 32px 32px 40px;
	margin: 24px 0px;
	text-decoration: none;

	:hover {
		border: 1px solid var(--card-link-hover);
		text-decoration: none;
	}
`;

const StyledTitle = styled.div`
	font-size: 24px;
	font-weight: 500;
	margin: 0px;
`;

const StyledBody = styled.p`
	font-size: 16px;
	margin-bottom: 16px;
	margin-right: 32px;
`;

const StyledBottom = styled.div`
	display: flex;
	justify-content: flex-end;
`;

interface Props {
	children: string;
	title: string;
	to: string;
}

export const Card = ({ children, title, to }: Props): JSX.Element => (
	<StyledContainer href={to}>
		<StyledTitle>{title}</StyledTitle>
		<StyledBody>{children}</StyledBody>
		<StyledBottom>
			<CardArrow />
		</StyledBottom>
	</StyledContainer>
);

export default Card;
