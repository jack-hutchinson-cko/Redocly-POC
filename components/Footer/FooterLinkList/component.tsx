import React, { useState } from 'react';
import { Chevron } from '../../svg';
import {
	Container,
	SolutionsLinksContainer,
	LinksContainer,
	ColMainWrapper,
	ColWrapper,
	FooterTitle,
	FooterLink,
	LinkText,
	StyledIcon,
} from './styles';

const FooterLinkList = ({ links = [] }) => {
	const [showSolutions, setSolutions] = useState<boolean>(false);
	const [showCompany, setCompany] = useState<boolean>(false);
	const [showInsights, setInsights] = useState<boolean>(false);
	const [showResources, setResources] = useState<boolean>(false);

	const setFuncs = [
		() => setCompany(!showCompany),
		() => setInsights(!showInsights),
		() => setResources(!showResources),
	];
	const showStatements = [showCompany, showInsights, showResources];

	return (
		<Container>
			{links?.map((link, i) => {
				if (i === 0) {
					return (
						<SolutionsLinksContainer key={i} onClick={() => setSolutions(!showSolutions)}>
							<FooterTitle>
								{link.primary.title[0].text}
								<StyledIcon>
									<Chevron />
								</StyledIcon>
							</FooterTitle>
							<ColMainWrapper>
								<ColWrapper isDisplayed={showSolutions}>
									{link.items.slice(0, link.items.length / 2 + 1).map((item, i) => (
										<FooterLink href={item.link_href[0].text} labelHere={item.link_name[0].text === 'Careers'} key={i}>
											<LinkText> {item.link_name[0].text} </LinkText>
										</FooterLink>
									))}
								</ColWrapper>

								<ColWrapper isDisplayed={showSolutions}>
									{link.items.slice(link.items.length / 2 + 1, link.items.length).map((item, i) => (
										<FooterLink href={item.link_href[0].text} labelHere={item.link_name[0].text === 'Careers'} key={i}>
											<LinkText> {item.link_name[0].text} </LinkText>
										</FooterLink>
									))}
								</ColWrapper>
							</ColMainWrapper>
						</SolutionsLinksContainer>
					);
				} else {
					return (
						<LinksContainer key={i} onClick={setFuncs[i - 1]}>
							<FooterTitle>
								{link.primary.title[0].text}
								<StyledIcon>
									<Chevron />
								</StyledIcon>
							</FooterTitle>
							<ColWrapper isDisplayed={showStatements[i - 1]}>
								{link.items.map((item, i) => (
									<FooterLink href={item.link_href[0].text} labelHere={item.link_name[0].text === 'Careers'} key={i}>
										<LinkText> {item.link_name[0].text} </LinkText>
									</FooterLink>
								))}
							</ColWrapper>
						</LinksContainer>
					);
				}
			})}
		</Container>
	);
};

export default FooterLinkList;
