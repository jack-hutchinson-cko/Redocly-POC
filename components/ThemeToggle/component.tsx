import React, { useEffect, useState } from 'react';
import styled, { css } from 'styled-components';

const StyledContainer = styled.div`
	background: var(--toggle-color);
	border-radius: 30px;
	cursor: pointer;
	display: flex;
	height: 24px;
	margin: auto 0;
	position: relative;
	top: -1px;
	transition: background 0.5s;
	user-select: none;
	width: 50px;
	z-index: 100;
`;

const SunAndMoonStyles = css`
	align-items: center;
	display: flex;
	font-size: 12px;
	padding: 0 4px;
	width: 50%;
`;

const StyledMoon = styled.div`
	${SunAndMoonStyles};
	justify-content: flex-end;
`;

const StyledSun = styled.div`
	${SunAndMoonStyles};
`;

const StyledThumb = styled.div`
	background: var(--toggle-circle);
	border-radius: 50%;
	height: 20px;
	left: 2px;
	position: absolute;
	top: 2px;
	transition: all 0.5s cubic-bezier(0.23, 1, 0.32, 1);
	width: 20px;
	transform: ${(props: { isDarkMode: boolean }) => (props.isDarkMode ? 'translateX(26px)' : '')};
`;

export const ThemeToggle = (): JSX.Element => {
	const [isDarkMode, setDarkMode] = useState<boolean>(false);

	useEffect(() => {
		if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
			document.getElementById('___gatsby').className = 'dark';
			setDarkMode(true);
		} else {
			setDarkMode(false);
		}
	}, []);

	const toggleTheme = () => {
		const activeTheme = !isDarkMode ? 'dark' : 'light';
		window.sessionStorage.setItem('theme', activeTheme);
		document.getElementById('___gatsby').className = activeTheme;
		setDarkMode(!isDarkMode);
	};

	return (
		<StyledContainer onClick={() => toggleTheme()}>
			<StyledMoon>🌛</StyledMoon>
			<StyledSun>☀️</StyledSun>
			<StyledThumb isDarkMode={isDarkMode}></StyledThumb>
		</StyledContainer>
	);
};

export default ThemeToggle;
export { StyledContainer, StyledThumb };
