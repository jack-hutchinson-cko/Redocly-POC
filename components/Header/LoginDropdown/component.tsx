import React from 'react';
import {
	StyledApplyLink,
	StyledBottom,
	StyledButtonContainer,
	StyledContainer,
	StyledTop,
	StyledLabel,
	StyledLink,
	StyledText,
	StyledTextBottom,
} from './loginDropdown.styles';
import { ArrowRight, TheHub, TestAccount } from '../../svg';
import ButtonLink from '../../ButtonLink';

const LoginDropdown = (): JSX.Element => (
	<StyledContainer>
		<StyledTop>
			<StyledLabel>
				<TheHub />
				The Hub
			</StyledLabel>
			<StyledText>Monitor transactions, business performance and customer trends.</StyledText>
			<StyledButtonContainer>
				<ButtonLink to="https://hub.checkout.com/login" hierarchy="secondary">
					Log In
				</ButtonLink>
				<StyledText>or</StyledText>
				<StyledApplyLink href="https://www.checkout.com/get-test-account">
					Apply for an account
					<ArrowRight />
				</StyledApplyLink>
			</StyledButtonContainer>
		</StyledTop>
		<StyledBottom>
			<StyledLabel>
				<TestAccount />
				Test account
			</StyledLabel>
			<StyledTextBottom>
				<StyledLink href="https://sandbox.checkout.com/login">Log in</StyledLink>
				or
				<StyledLink href="https://www.checkout.com/get-test-account">apply for a test account</StyledLink>
			</StyledTextBottom>
		</StyledBottom>
	</StyledContainer>
);

export default LoginDropdown;
