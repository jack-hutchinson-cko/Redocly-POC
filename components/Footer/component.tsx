import React from 'react';
import { Section } from './styles';
import { Background, LogoContainer, LogoWrapper, LineSeparator } from './styles';
import { CheckoutLogo } from '../svg/CheckoutLogo';
import FooterLinkList from './FooterLinkList';
import SubFooter from './SubFooter';
import { dataFooter } from './footerData';

const Footer = () => (
	<Background>
		<Section>
			<LogoContainer>
				<LogoWrapper>
					<CheckoutLogo />
				</LogoWrapper>
			</LogoContainer>

			<FooterLinkList links={dataFooter.body} />

			<LineSeparator />

			<SubFooter />
		</Section>
	</Background>
);

export default Footer;
