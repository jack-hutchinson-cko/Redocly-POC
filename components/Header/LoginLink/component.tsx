import React, { useState } from 'react';
import { Chevron } from '../../svg';
import { StyledLink } from '../styles';
import { StyledContainer, StyledLoginLink, StyledSVG } from './loginLink.styles';
import LoginDropdown from '../LoginDropdown';

const LoginLink = (): JSX.Element => {
	const [isOpen, setOpen] = useState<boolean>(false);

	return (
		<StyledLoginLink>
			<StyledContainer onClick={() => setOpen(!isOpen)}>
				<StyledLink>Log in</StyledLink>
				<StyledSVG isOpen={isOpen}>
					<Chevron />
				</StyledSVG>
			</StyledContainer>
			{isOpen && <LoginDropdown />}
		</StyledLoginLink>
	);
};

export default LoginLink;
export { StyledContainer };
