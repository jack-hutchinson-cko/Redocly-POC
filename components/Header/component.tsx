import React from 'react';
import { ThemeToggle } from '../index';
import LoginLink from './LoginLink';
import MenuLink from './MenuLink';
import {
	StyledContainer,
	StyledLink,
	StyledLeftHeader,
	StyledLogo,
	StyledLogoIcon,
	StyledLogoText,
	StyledRightHeader,
} from './styles';

const Header = (): JSX.Element => (
	<StyledContainer>
		<StyledLeftHeader>
			<StyledLogo href="/">
				<StyledLogoIcon />
				<StyledLogoText>Documentation</StyledLogoText>
			</StyledLogo>
			<MenuLink />
		</StyledLeftHeader>
		<StyledRightHeader>
			<LoginLink />
			<StyledLink>Get test account</StyledLink>
			<ThemeToggle />
		</StyledRightHeader>
	</StyledContainer>
);

export default Header;
