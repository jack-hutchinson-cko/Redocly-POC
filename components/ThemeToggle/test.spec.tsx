import { shallow } from 'enzyme';
import React from 'react';
import { ThemeToggle, StyledContainer, StyledThumb } from './component';

describe('ThemeToggle component', () => {
	beforeEach(() => {
		global.window = {
			matchMedia: jest.fn(),
			sessionStorage: {
				setItem: jest.fn(),
				getItem: jest.fn(),
				removeItem: jest.fn(),
				clear: jest.fn(),
				key: jest.fn(),
				length: 2,
			},
		};

		global.document = {
			getElementById: jest.fn(() => ({ className: '' })),
		};
	});

	afterEach(() => {
		delete global.window;
	});

	test('component matches snapshot', () => {
		expect.assertions(1);

		const wrapper = shallow(<ThemeToggle />);

		expect(wrapper).toMatchSnapshot();
	});

	test('isDarkMode is true on container click', () => {
		expect.assertions(2);

		const wrapper = shallow(<ThemeToggle />);
		wrapper.find(StyledContainer).simulate('click');
		const thumbDarkMode = wrapper.find(StyledThumb).props().isDarkMode;

		expect(wrapper).toMatchSnapshot();
		expect(thumbDarkMode).toBe(true);
	});
});
