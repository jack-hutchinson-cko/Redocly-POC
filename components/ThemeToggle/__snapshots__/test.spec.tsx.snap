// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeToggle component component matches snapshot 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.div>
    🌛
  </styled.div>
  <styled.div>
    ☀️
  </styled.div>
  <styled.div
    isDarkMode={false}
  />
</styled.div>
`;

exports[`ThemeToggle component isDarkMode is true on container click 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.div>
    🌛
  </styled.div>
  <styled.div>
    ☀️
  </styled.div>
  <styled.div
    isDarkMode={true}
  />
</styled.div>
`;
